### PR TITLE
feat(@ai-sdk/azure): Adds embedding support for Azure OpenAI provider

### PR DESCRIPTION
**Pull request by Und3rf10w**
Adds embedding support for the Azure OpenAI provider.

Please note that this PR includes patches to the @ai-sd/openai provider as well.

### CODE REVIEW

File: packages/azure/src/azure-openai-provider.ts [hunk 1]
```diff
@@ -80,9 +98,23 @@ export function createAzure(
...
+        `https://${getResourceName()}.openai.azure.com/openai/deployments/${modelId}${path}?api-version=2024-05-01-preview`,
+      compatibility: 'compatible',
+      fetch: options.fetch,
+    });
+
   const provider = function (
     deploymentId: string,
-    settings?: OpenAIChatSettings,
+    type: 'chat' | 'completion' | 'embeddings' = 'chat',
+    settings?: OpenAIChatSettings | OpenAIEmbeddingSettings,
```
Comment (lgrammel) [inline packages/azure/src/azure-openai-provider.ts:117]: the main entry point should just be for the main language model, i.e., the above 2 lines can be reverted

Comment (Und3rf10w) [reply packages/azure/src/azure-openai-provider.ts]: Resolved in: 4fd114787b4ed8527734d46ac5e9a2786cebbc49

File: packages/azure/src/azure-openai-provider.ts [hunk 2]
```diff
@@ -80,9 +98,23 @@ export function createAzure(
...
 
+  const createEmbeddingModel = (
+    modelId: string,
+    settings: OpenAIEmbeddingSettings = {},
+  ) =>
+    new OpenAIEmbeddingModel(modeId, settings, {
+      provider: 'azure-openai.embeddings',
+      headers: getHeaders,
+      url: ({ path }) =>
+        `https://${getResourceName()}.openai.azure.com/openai/deployments/${modelId}${path}?api-version=2024-05-01-preview`,
```
[the remark this answers is not in the document]
Comment (Und3rf10w) [reply packages/azure/src/azure-openai-provider.ts]: I'm not sure I entirely follow here, basically abstract that out to a function?

Also, would you ever want user(s) to provide the `api-version` to use themselves? Would you want to include something option for that, and default to `2024-05-01-preview`?

**Comment by lgrammel** [comment]
Looks great, I'll create a new PR with docs etc.

**Comment by lgrammel** [comment]
superseeded by https://github.com/vercel/ai/pull/1994

**Pull request closed without merging**